workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/ViewRegistry.kt

```diff
@@ -112,6 +112,12 @@ public fun ViewRegistry(): ViewRegistry = TypedViewRegistry()
  * If that returns null, falls back to the factory provided by the rendering's
  * implementation of [AndroidViewRendering.viewFactory], if there is one.
  *
+ * The returned view will have a [WorkflowLifecycleOwner] set on it, the caller _must_ ensure the
```

Comment:
What would you think about adding a new optional param to `buildView` to handle the bookending:

```kotlin
public fun <RenderingT : Any> ViewRegistry.buildView(
  initialRendering: RenderingT,
  initialViewEnvironment: ViewEnvironment,
  contextForNewView: Context,
  container: ViewGroup? = null,
  replaces: View? = null
): View
```

It's a shame the way it kind of collides with `container`, don't like it.

Could also just add a new extension, and derive most of the noise from `previous`:

```kotlin
public fun <RenderingT : Any> ViewRegistry.replaceView(
  initialRendering: RenderingT,
  initialViewEnvironment: ViewEnvironment,
  previous: View? = null
): View
```

Reply:
I don't think either of those would work with `ModalContainer`, which both needs to be able to independently destroy any of it's modal children (not replace them), and delegate the `buildView` call to an abstract method.